clippy_lints/src/suspicious_xor_used_as_pow.rs

```diff
@@ -34,21 +36,20 @@ impl LateLintPass<'_> for ConfusingXorAndPow {
             left.span.ctxt() == right.span.ctxt() &&
             let ExprKind::Lit(lit_left) = &left.kind &&
             let ExprKind::Lit(lit_right) = &right.kind &&
-            let snip_left = snippet_with_context(cx, lit_left.span, lit_left.span.ctxt(), "..", &mut Applicability::MaybeIncorrect) &&
-            let snip_right = snippet_with_context(cx, lit_right.span, lit_right.span.ctxt(), "..", &mut Applicability::MaybeIncorrect) &&
-            let Some(left_val) = NumericLiteral::from_lit_kind(&snip_left.0, &lit_left.node) &&
-            let Some(right_val) = NumericLiteral::from_lit_kind(&snip_right.0, &lit_right.node) &&
-			left_val.is_decimal() &&
-			right_val.is_decimal() {
-					clippy_utils::diagnostics::span_lint_and_sugg(
-					        cx,
-					        SUSPICIOUS_XOR_USED_AS_POW,
-					        expr.span,
-					        "`^` is not the exponentiation operator",
-					        "did you mean to write",
-					        format!("{}.pow({})", left_val.format(), right_val.format()),
-					        Applicability::MaybeIncorrect,
-					    );
+            matches!(lit_right.node, LitKind::Int(..) | LitKind::Float(..)) &&
+            matches!(lit_left.node, LitKind::Int(..) | LitKind::Float(..)) &&
+            NumericLiteral::from_lit_kind(&snippet(cx, lit_right.span, ".."), &lit_right.node).is_some_and(|x| x.is_decimal())
+            {
+                span_lint_and_sugg(
+                    cx,
+                    SUSPICIOUS_XOR_USED_AS_POW,
+                    expr.span,
+                    "`^` is not the exponentiation operator",
+                    "did you mean to write",
+                    format!("{}.pow({})", lit_left
+                    .node, lit_right.node),
```

Comment:
NIT: This format is a little weird. rustfmt sadly doesn't work in functions that have `if-let` chains in them. Can you fix this?